src/utils.jl

```diff
@@ -29,13 +29,15 @@ parts of a composition.
 """
 function compose(data, cols::Vector{Symbol})
   # non-compositional columns
-  result = data[:,setdiff(names(data), cols)]
+  result = data[:,setdiff(propertynames(data), cols)]
 
   # compositional columns
   coda = map(eachrow(data[:,cols])) do row
-    Composition(Tuple(names(row)), values(row))
+    Composition(Tuple(propertynames(row)), values(row))
   end
-  result[Symbol(join(cols, "|"))] = coda
+  result[!, Symbol(join(cols, "|"))] = coda
 
   result
 end
+
+compose(data, cols::Array{String,1}) = compose(data, Symbol.(cols))
```

Comment:
We can simplify this as `cols::Vector{String}` given that `Vector{T} = Array{T,1}`.

Reply:
Or even better, replace it with a `NTuple{N,String}` as discussed above.

Reply:
Ok! 
Added NTuple -> Array conversion, using `collect(cols)` for DataFrame column selection.